src/goto-symex/goto_state.h

```diff
@@ -63,6 +63,13 @@ class goto_statet
   /// Threads
   unsigned atomic_section_id = 0;
 
+  /// Get the currently calculated complexity for this state. Used
+  /// to short-circuit states if they have become too computationally complex.
+  inline std::size_t get_complexity()
+  {
+    return guard.as_expr().size();
```

Comment:
:bulb: this would be costly for BDDs. Ideally we should have a `complexity` method for guards, so that it can be substituted to something more efficient (you don't need to do it in this PR)

Reply:
Yeah, it was something I was musing on and was going to talk with you at some point or do a little bit more poking around to find a proper solution.

Reply:
I think the proper solution for BDDs would be to use something like the dagSize method of Cudd http://web.mit.edu/sage/export/tmp/y/usr/share/doc/polybori/cudd/cuddAllDet.html#Cudd_DagSize
but I'm not sure there is something similar for miniBDD

Reply:
I would add a comment here saying that if we start using BDDs for guards then we should use something more efficient, like dagSize.